cli/tests/test_diffedit_command.rs

```diff
@@ -514,3 +516,119 @@ fn test_diffedit_old_restore_interactive_tests() {
     +unrelated
     "###);
 }
+
+#[test]
+fn test_diffedit_restore_descendants() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file"), "println!(\"foo\")\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file"), "println!(\"bar\")\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file"), "println!(\"baz\");\n").unwrap();
+
+    let edit_script = test_env.set_up_fake_diff_editor();
+
+    // Add a ";" after the line with "bar". There should be no conflict.
+    std::fs::write(edit_script, "write file\nprintln!(\"bar\");\n").unwrap();
+    let (stdout, stderr) = test_env.jj_cmd_ok(
+        &repo_path,
+        &["diffedit", "-r", "@-", "--restore-descendants"],
+    );
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r#"
+    Created rlvkpnrz 62b8c2ce (no description set)
+    Rewritten 1 descendant commits
+    Working copy now at: kkmpptxz 321d1cd1 (no description set)
+    Parent commit      : rlvkpnrz 62b8c2ce (no description set)
+    "#);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["diff", "--git"]);
+    insta::assert_snapshot!(stdout, @r#"
+    diff --git a/file b/file
+    index 1a598a8fc9..7b6a85ab5a 100644
+    --- a/file
+    +++ b/file
+    @@ -1,1 +1,1 @@
+    -println!("bar");
+    +println!("baz");
+    "#);
+}
+
+#[test]
+fn test_diffedit_restore_descendants_with_multiple_parents() {
```

Comment:
This test should ideally go into the lib crate, testing the new function in the first commit in this PR. It makes sense to also have a test of the CLI feature, but we should not have to test more than just the basics here. The lib tests are faster, and they mean that we test the reusable functionality, so we don't have to test it for every command that gains a `--restore-descendants` flag. Sorry that I didn't notice this earlier.

Reply:
I won't be able to do it right away, but I'll keep this on my list.

Reply:
No problem. If you think you'll get to it soon, please update the PR. If it might take a bit longer (than a few days, says), feel free to send another PR instead.

Reply:
I've added a new test in its own commit

Reply:
Thanks! I think you can delete this test now, unless you think it covers some case that the jj-lib test and the test above don't cover.